src/assisted_test_infra/test_infra/controllers/ipxe_controller/ipxe_controller.py

```diff
@@ -0,0 +1,58 @@
+import os
+import shutil
+
+import consts
+from assisted_test_infra.test_infra import utils
+from service_client import log
+
+
+class IPXEController:
+    def __init__(
+        self,
+        name=None,
+        port=consts.DEFAULT_IPXE_SERVER_PORT,
+        ip=consts.DEFAULT_IPXE_SERVER_IP,
+    ):
+        self.name = name
+        self.ip = ip
+        self.port = port
+        self.dir = os.path.dirname(os.path.realpath(__file__))
+        self.ipxe_scripts_folder = f"{self.dir}/server/ipxe_scripts"
+
+    def remove(self):
+        log.info(f"Removing iPXE Server {self.name}")
+        utils.remove_running_container(container_name=self.name)
+        self._remove_ipxe_scripts_folder()
+
+    def start(self, api_client, infra_env_id, cluster_name):
```

Comment:
Maybe `api_client` can be part of the class properties

Reply:
Good idea, made it part of the class property. Thanks!